crates/uv-python/src/installation.rs

```diff
@@ -484,7 +484,33 @@ impl Ord for PythonInstallationKey {
             .then_with(|| self.version().cmp(&other.version()))
             .then_with(|| self.os.to_string().cmp(&other.os.to_string()))
             // Architectures are sorted in preferred order, with native architectures first
-            .then_with(|| self.arch.cmp(&other.arch).reverse())
+            .then_with(|| {
+                // For the time being, manually sort windows to make aarch64 windows disfavored
```

Comment:
Why not implement this in https://github.com/astral-sh/uv/blob/546a6d017d6294712a9796630397862ba86d77ff/crates/uv-python/src/platform.rs#L40-L59 instead under a Windows cfg? Do we need the `self.os` check here?

Reply:
(this just looks a lot like the logic already in `Arch::cmp` and I think it'd be easier to reason about in a single location)

Reply:
Yeah it's because of the os check. Not that it's super useful now that I think about it harder, but it feels... wrong... that we would resolve a request for `python-3.11-darwin-any` to intel on arm64 windows.

Reply:
Hm... it does feel wrong... but like... not a real use-case today. I worry about this being brittle. I'll ponder this overnight.

It does suggest that long-term we need a dedicated `Platform` type to encapsulate both the OS and the Arch and implement `Ord` on that instead?